### PR TITLE
Add support for subcriptions

### PR DESCRIPTION
**Pull request by victorhmp**
#### What is the purpose of this pull request?
Add support for items with subscriptions attached to them.

#### What problem is this solving?
Subscriptions would be treated as a normal attachment and the text displayed would not be formatted for the user.

#### How should this be manually tested?

`cd react && yarn test`

#### Screenshots or example usage

screen shot 2019-02-11 at 5 28 42 pm

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/ProductList/Attachment.tsx [hunk 1]
```diff
@@ -10,6 +10,66 @@ interface Props {
   bundleInfo: Bundle[]
 }
 
+const getSubscriptionInfo = (
```
Comment (klzns) [inline react/components/ProductList/Attachment.tsx]: I would keep this in a separate module and add tests to it.

File: react/__mocks__/SubsMockComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  const {
+    isSubscription,
+    subsFrequency,
+    subsPurchaseDay,
+    subsValidityBegin,
+    subsValidityEnd,
+  } = getSubscriptionInfo(attachmentItem, intl)
+
+  return (
+    <ul>
```
Comment (klzns) [inline react/__mocks__/SubsMockComponent.tsx]: You can change it a render props component


```js

return children({
    isSubscription,
    subsFrequency,
    subsPurchaseDay,
    subsValidityBegin,
    subsValidityEnd,
  })
```

Then, at the tests you may pass a mock function and check the params that were called to it:


```js
const fn = jest.mock()

render(
<SubscriptionMock
  attachmentItem={attachmentItem}
  children={fn} />
)

const params = fn.mock.calls[0][0]

expected(params.isSubscription).toBe(true)
expected(params.subsFrequency).toBe('foo')
// etc
```

Comment (victorhmp) [reply react/__mocks__/SubsMockComponent.tsx]: Yep! Much cleaner hahahah
Just one thing, I could not return `children({...})` directly, so I called the mock function from inside the mock component and after that I return `null`, jest wouldn't let me return "nothing" in my component, as it assumed `fn` was.

File: react/utils/index.tsx [hunk 3]
```diff
@@ -21,3 +21,63 @@ export interface PaymentGroupInfo {
   url: string | null
   value: number
 }
+
+export const getSubscriptionInfo = (
```
Comment (salesfelipe) [inline react/utils/index.tsx]: this looks more like two different functions to me

File: react/components/ProductList/Attachment.tsx [hunk 4]
```diff
@@ -73,42 +74,65 @@ const ProductAttachment: FunctionComponent<Props & InjectedIntlProps> = ({
...
-          <article
-            className="bg-muted-5 pv3 ph5 br2 mv4"
-            key={attachmentItem.name}
-          >
-            <div className="flex justify-between">
-              <p className="block c-on-base">{attachmentItem.name}</p>
-              <div className="flex items-center">
-                <p className="mr5">
-                  {intl.formatMessage({ id: 'order.totals.pickup.free' })}
+        attachmentsInfo.map(attachmentItem => {
```
[the remark this answers is not in the document]
Comment (victorhmp) [reply react/components/ProductList/Attachment.tsx]: Removed logic for subscriptions to a new component.

File: react/utils/index.tsx (hunk 3 above)
Comment (victorhmp) [reply react/utils/index.tsx]: Created new function `isSubscription` that checks if a given `Attachment` is a subscription.

**Pull request merged**